src/components/pagination/pagination.e2e.ts

```diff
@@ -179,4 +180,81 @@ describe("calcite-pagination", () => {
       expect(toggleSpy).toHaveReceivedEventTimes(0);
     });
   });
+
+  describe("number locale support", () => {
+    let page: E2EPage;
+    let element: E2EElement;
+    let noSeparator: string[];
+    let withSeparator: string[];
+    let getDisplayedValuesArray;
+    const expectedNotSeparatedValueArray: string[] = ["14999997", "14999998", "14999999", "15000000"];
+
+    const formattedValuesPerLanguageObject = {
+      "de-CH": ["14’999’997", "14’999’998", "14’999’999", "15’000’000"],
+      en: ["14,999,997", "14,999,998", "14,999,999", "15,000,000"],
+      es: ["14.999.997", "14.999.998", "14.999.999", "15.000.000"],
+      fr: ["14 999 997", "14 999 998", "14 999 999", "15 000 000"],
+      hi: ["1,49,99,997", "1,49,99,998", "1,49,99,999", "1,50,00,000"]
+    };
+
+    beforeEach(async () => {
+      page = await newE2EPage();
+      await page.setContent(
+        html`<calcite-pagination lang="en" group-separator total="150000000" num="10"></calcite-pagination>`
+      );
+      element = await page.find("calcite-pagination");
+
+      // assign temporary class to the last displayed page to query outside of evaluate
+      await page.evaluate(() => {
+        const el = document.querySelector("calcite-pagination");
+        const lastButton = Array.from(el.shadowRoot.querySelectorAll("button")).find((el) => {
+          return el.innerText === "15,000,000";
+        });
+        lastButton.classList.add("lastPage");
+      });
+      const lastButton = await page.find(`calcite-pagination >>> .lastPage`);
+      await lastButton.click();
+
+      const buttonListAbridged = await (await page.findAll(`calcite-pagination >>> .${CSS.page}`)).slice(-4);
```

Comment:
I'd suggest creating a utility function inside this `describe` block that other tests can reference within the node testing context. For example:

```ts
async function getDisplayedValues(): Promise<string[]> {
  const buttons = await page.findAll("calcite-pagination >>> .page");
  const buttonsWithGroupSeparator = buttons.slice(-4);
  return buttonsWithGroupSeparator.map(button => button.innerText);
}

// within it("something", async () => {

expect(noSeparator).toEqual(
  // tests can use this method directly without having to run it within the browser context
  await getDisplayedValues()
); 
```